runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -11955,6 +11957,124 @@ J9::X86::TreeEvaluator::inlineStringLatin1Inflate(TR::Node *node, TR::CodeGenera
+   TR::LabelSymbol *startMainLoop = generateLabelSymbol(cg);
+   TR::LabelSymbol *endMainLoop = generateLabelSymbol(cg);
+   TR::LabelSymbol *startResidualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *endResidualLabel = generateLabelSymbol(cg);
+
+   int maxDepCount = 10;
+   bool is64bit = cg->comp()->target().is64Bit();
+   TR::RegisterDependencyConditions *deps = generateRegisterDependencyConditions(0, maxDepCount, cg);
+
+   deps->addPostCondition(bufReg, TR::RealRegister::eax, cg);
+   deps->addPostCondition(offsetReg, TR::RealRegister::ebx, cg);
+   deps->addPostCondition(tmpReg, TR::RealRegister::edi, cg);
+   deps->addPostCondition(result, TR::RealRegister::ecx, cg);
+   deps->addPostCondition(endReg, TR::RealRegister::NoReg, cg);
+   deps->addPostCondition(tmp2Reg, TR::RealRegister::NoReg, cg);
+
+   ///
+   generateLabelInstruction(TR::InstOpCode::label, node, startMain2Loop, cg);
+   generateRegRegInstruction(TR::InstOpCode::MOVRegReg(is64bit), node, tmpReg, offsetReg, cg);
+   generateRegImmInstruction(TR::InstOpCode::ADD4RegImm4, node, tmpReg, 64, cg);
+
+   generateRegRegInstruction(TR::InstOpCode::CMP4RegReg, node, tmpReg, endReg, cg);
+   generateLabelInstruction(TR::InstOpCode::JG4, node, endMain2Loop, cg);
+
+   // Main Loop Body
+
+   for (int i = 0; i < 8; i++)
+      {
+      TR::MemoryReference *mr = generateX86MemoryReference(bufReg, offsetReg, 0, TR::Compiler->om.contiguousArrayHeaderSizeInBytes() + i * 8, cg);
+      generateRegMemInstruction(TR::InstOpCode::CRC32QRegReg, node, result, mr, cg);
+      }
+
+   generateRegImmInstruction(TR::InstOpCode::ADD4RegImm4, node, offsetReg, 64, cg);
+
+   // Loop back
+   generateLabelInstruction(TR::InstOpCode::JMP4, node, startMain2Loop, cg);
+   generateLabelInstruction(TR::InstOpCode::label, node, endMain2Loop, cg);
+   ///
+
+   generateLabelInstruction(TR::InstOpCode::label, node, startMainLoop, cg);
+   generateRegRegInstruction(TR::InstOpCode::MOVRegReg(is64bit), node, tmpReg, offsetReg, cg);
+   generateRegImmInstruction(TR::InstOpCode::ADD4RegImm4, node, tmpReg, 8, cg);
+
+   generateRegRegInstruction(TR::InstOpCode::CMP4RegReg, node, tmpReg, endReg, cg);
+   generateLabelInstruction(TR::InstOpCode::JG4, node, endMainLoop, cg);
+
+   // Main Loop Body
+
+   TR::MemoryReference *mr = generateX86MemoryReference(bufReg, offsetReg, 0, TR::Compiler->om.contiguousArrayHeaderSizeInBytes(), cg);
+   generateRegMemInstruction(TR::InstOpCode::CRC32QRegReg, node, result, mr, cg);
+
+   generateRegImmInstruction(TR::InstOpCode::ADD4RegImm4, node, offsetReg, 8, cg);
+
+   // Loop back
+   generateLabelInstruction(TR::InstOpCode::JMP4, node, startMainLoop, cg);
+   generateLabelInstruction(TR::InstOpCode::label, node, endMainLoop, cg);
+
+   generateRegRegInstruction(TR::InstOpCode::MOVRegReg(is64bit), node, tmpReg, endReg, cg);
+   generateRegRegInstruction(TR::InstOpCode::SUBRegReg(is64bit), node, tmpReg, offsetReg, cg);
+
```

Comment:
If the intent here is to introduce some form of jump table to process the final residue then
1) You should have more comments describing what you're doing
2) I think it should be possible to come up with an instruction sequence that is denser (and doesn't involve incrementing offsetReg each time).  Would something like this work?  I just did this quickly off the top of my head so there are likely errors.
```
   ; assumes number of bytes to process is 0-7 and they must be done in order
   ;
   mov tempReg, endReg
   sub tempReg, offsetReg  ; number of bytes left to process
   lea offsetReg, [tempReg -7]  ; adjustment = length - 7  
   mov tempReg, [magic arithmetic to compute jump target based on length]
   jmp tempReg
   crc32b result, [bufRef + offsetReg + 0]
   crc32b result, [bufRef + offsetReg + 1]
   crc32b result, [bufRef + offsetReg + 2]
   crc32b result, [bufRef + offsetReg + 3]
   crc32b result, [bufRef + offsetReg + 4]
   crc32b result, [bufRef + offsetReg + 5]
   crc32b result, [bufRef + offsetReg + 6]
   crc32b result, [bufRef + offsetReg + 7]
```